test/distribute-lp-reward.js

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env node
+const tape = require('tape')
+
+const { bigNumberify } = require('ethers').utils
+
+const {
+	getDistributionForPeriodWithMultiplier,
+	NULL_ADDRESS,
+	parseUniswapTransferEvents,
+	parseBalancerTransferEvents
+} = require('../scripts/distribute-lp-rewards/lib')
+
+const sum = tree => Object.values(tree).reduce((a, b) => a.add(b), bigNumberify(0))
+
+const addDays = initialDate => days => {
+	const d = new Date(initialDate.toString())
+	d.setDate(initialDate.getDate() + days)
+	return d
+}
+
+tape('uniswap: distributes reward correctly', function(t) {
+	// 2 weeks between DISTRIBUTION_START & DISTRIBUTION_ENDS
+	const DISTRIBUTION_START = new Date()
+	DISTRIBUTION_START.setDate(DISTRIBUTION_START.getDate() - 15)
+
+	const DISTRIBUTION_ENDS = new Date()
+	DISTRIBUTION_ENDS.setDate(DISTRIBUTION_ENDS.getDate() - 1)
+
+	const addDaysToDistributionStart = addDays(DISTRIBUTION_START)
+
+	const testUniswapTransferEvents = [
+		{
+			time: Math.floor(DISTRIBUTION_START.getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x1'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(1).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x2'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(1) / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x4'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(7).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x5'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(9).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x6'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(9).getTime() / 1000),
+			values: {
+				from: '0x4',
+				value: bigNumberify('3'),
+				to: '0x7'
+			}
+		}
+	]
+
+	const liquidityDurationByMultiplier = {
+		1: 10, // 1 week => 0.1% increase
+		2: 20,
+		3: 30
+	}
+
+	const distributionStarts = Math.floor(DISTRIBUTION_START.getTime() / 1000)
+	const distributionEnds = DISTRIBUTION_ENDS.getTime() / 1000
```

Comment:
- `Math.floor`
- distribution params can be common for all the test as they are the same